sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -61,40 +62,65 @@ public abstract class WindowedValue<T> {
 
   /** Returns a {@code WindowedValue} with the given value, timestamp, and windows. */
   public static <T> WindowedValue<T> of(
-      T value, Instant timestamp, Collection<? extends BoundedWindow> windows, PaneInfo pane) {
+      T value,
+      Instant timestamp,
+      Collection<? extends BoundedWindow> windows,
+      PaneInfo pane,
+      @Nullable ElementMetadata elementMetadata) {
     checkArgument(pane != null, "WindowedValue requires PaneInfo, but it was null");
     checkArgument(windows.size() > 0, "WindowedValue requires windows, but there were none");
 
```

Comment:
Perhaps also validate here that elementMetadata is null if and only if PaneInfo expects metadata?